app/scripts/views/forms.coffee

```diff
@@ -971,6 +972,26 @@ define [
       @listenToNewFormView()
       @showNewFormViewAnimate()
 
+
+    # Request the history of the form model, the user can click on a revision and see the details
+    # preferably by simply showing the other revision as a model along side in the same views
+    fetchHistory: (formModel) ->
+
+      fielddbHelperModel = new (FieldDB.Datum)(_id: formModel.id)
+      fielddbHelperModel.fetchRevisions().then(((revisions) ->
+        formModel.set 'previous_versions', revisions.map((revisionUrl) ->
+          # TODO can we avoid fetching them until the user clicks on the one they want?
```

Comment:
@jrwdunham  this was my new question, would it be okay to avoid fetching hte previous versions until the user clicks on the one they actually want to see..

I dont know how to do any visuals in your app, so generating the history is "done", but i cant make the interface to show it. Ideally Id like the history view to have 2 from views, and the user can select the history item they want to see in each form view (kind of like you do in a wikipedia history)

![screen shot 2015-04-10 at 11 15 29 am](https://cloud.githubusercontent.com/assets/196199/7081823/eea0aa56-df72-11e4-990d-11c036e8262d.png)